src/utils/PublicIdentity/Encoding.ts

```diff
@@ -1,9 +1,25 @@
-import { IPublicIdentity } from '@kiltprotocol/sdk-js'
+import { IPublicIdentity, IDelegationNode } from '@kiltprotocol/sdk-js'
 
 const encodePublicIdentity = (publicIdentity: IPublicIdentity): string[] => [
   publicIdentity.address,
   publicIdentity.boxPublicKeyAsHex,
   ...(publicIdentity.serviceAddress ? [publicIdentity.serviceAddress] : []),
 ]
 
-export default encodePublicIdentity
+export type IAttesterWithDelegation = {
+  publicIdentity: IPublicIdentity
+  delegationId: IDelegationNode['id']
+}
+
+const encodePublicIdentityWithDelegation = (
+  publicIdentityWithDelegation: IAttesterWithDelegation
+): string[] => [
+  publicIdentityWithDelegation.publicIdentity.address,
+  publicIdentityWithDelegation.publicIdentity.boxPublicKeyAsHex,
+  ...(publicIdentityWithDelegation.publicIdentity.serviceAddress
+    ? [publicIdentityWithDelegation.publicIdentity.serviceAddress]
+    : []),
+  publicIdentityWithDelegation.delegationId,
```

Comment:
I think it is easier to put the delegationId at the top. Otherwise, you will have to check the array length, before you know at what index it is